sdk/samples/iot/hub/src/paho_iot_hub_methods_example.c

```diff
@@ -61,62 +61,99 @@ static az_span ping_method_fail_response = AZ_SPAN_LITERAL_FROM_STR("{}");
 static az_iot_hub_client client;
 static MQTTClient mqtt_client;
 
-// Read OS environment variables using stdlib function
+//
+// Configuration and connection functions - uses MQTT API
+//
+static az_result read_configuration_and_init_client();
 static az_result read_configuration_entry(
     const char* name,
     const char* env_name,
     char* default_value,
     bool hide_value,
     az_span buffer,
-    az_span* out_value)
+    az_span* out_value);
+static az_result create_mqtt_endpoint(char* destination, int32_t destination_size, az_span iot_hub);
+static int connect_device();
+static int subscribe();
+
+//
+// Messaging functions
```

Comment:
Yea same here like a previous comment. This does use MQTT API so we might need to be more specific here.

I think for now, it might be enough to remove the "uses MQTT API" part until we structure this a little better (to distinguish between MQTT stack functionality and IoT Hub client stuff.

Reply:
sounds like a plan